### ISSUE TITLE
System tray icon not working in LXDE

### ISSUE THREAD

**Issue by ferdnyc** [BiglySoftware/BiglyBT#570]
For quite some time now (possibly from the very beginning), BiglyBT hasn't shown a system tray icon in my Linux setup (LXDE desktop, running in a VLC session on a headless fileserver), even though the tray icon is enabled in Options > Interface.

The console logs these messages, apparently whenever it _tries_ to show the tray icon:
```
[06:17:03.059] {stderr} java.lang.NullPointerException
[06:17:03.059] {stderr} 	at com.biglybt.ui.swt.systray.TrayItemDork.setMenu(TrayItemDork.java:112)
[06:17:03.059] {stderr} 	at com.biglybt.ui.swt.systray.SystemTraySWT.updateUI(SystemTraySWT.java:599)
[06:17:03.059] {stderr} 	at com.biglybt.ui.swt.uiupdater.UIUpdaterSWT.update(UIUpdaterSWT.java:296)
[06:17:03.060] {stderr} 	at com.biglybt.ui.swt.uiupdater.UIUpdaterSWT.access$5(UIUpdaterSWT.java:281)
[06:17:03.060] {stderr} 	at com.biglybt.ui.swt.uiupdater.UIUpdaterSWT$1.runWithDisplay(UIUpdaterSWT.java:159)
[06:17:03.060] {stderr} 	at com.biglybt.ui.swt.utils.SWTRunnable.run(SWTRunnable.java:37)
[06:17:03.060] {stderr} 	at org.eclipse.swt.widgets.RunnableLock.run(RunnableLock.java:37)
[06:17:03.060] {stderr} 	at org.eclipse.swt.widgets.Synchronizer.runAsyncMessages(Synchronizer.java:182)
[06:17:03.060] {stderr} 	at org.eclipse.swt.widgets.Display.runAsyncMessages(Display.java:4497)
[06:17:03.060] {stderr} 	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:4110)
[06:17:03.060] {stderr} 	at com.biglybt.ui.swt.mainwindow.SWTThread.<init>(SWTThread.java:284)
[06:17:03.060] {stderr} 	at com.biglybt.ui.swt.mainwindow.SWTThread.createInstance(SWTThread.java:64)
[06:17:03.060] {stderr} 	at com.biglybt.ui.swt.Initializer.<init>(Initializer.java:179)
[06:17:03.060] {stderr} 	at com.biglybt.ui.swt.UI.takeMainThread(UI.java:110)
[06:17:03.060] {stderr} 	at com.biglybt.ui.Main.main(Main.java:236)
[06:17:03.060] {stderr} 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[06:17:03.060] {stderr} 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[06:17:03.060] {stderr} 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[06:17:03.060] {stderr} 	at java.lang.reflect.Method.invoke(Method.java:498)
[06:17:03.060] {stderr} 	at com.biglybt.launcher.MainExecutor$1.run(MainExecutor.java:37)
[06:17:03.060] {stderr} 	at java.lang.Thread.run(Thread.java:748)
```

#### System details
```
Java 1.8.0_181 (64 bit)
  Oracle Corporation
/usr/lib/jvm/java-1.8.0-openjdk-1.8.0.181-7.b13.fc28.x86_64/jre

SWT v4757, gtk/3.22.30
Linux v4.17.12-200.fc28.x86_64, amd64 (64 bit)
B1.6.0.1_B02/4 az3
```

**Comment by TuxPaper** [comment]
Are there any other console lines during startup that have the word "SystemTray" in it?  It would probably be right at startup.  The library we use should display some error (maybe something about JNA)

It's possible the library doesn't support an LXDE desktop, however, it should be compatible with most systems supporting gtk, which LXDE appears to be.

**Comment by ferdnyc** [comment]
Yes, a very similar one, just four lines from the start of logging for the session. Here's the whole thing from the point the logfile is opened:
```
[8/11/18 1:30 AM] Log File Opened for BiglyBT 1.6.0.1_B03^M
[01:30:46] [stderr] DEBUG::Sat Aug 11 01:30:46 EDT 2018::com.biglybt.core.util.AEDiagnostics::markDirty::377:
[01:30:46] [stderr]   Marking as dirty
[01:30:46] [stderr]     <init> (CoreImpl.java:269), create (CoreImpl.java:170), create (CoreFactory.java:43), main (Main.java:189), invoke0 (NativeMethodAccessorImpl.java:-2), invoke (NativeMethodAccessorImpl.java:62), invoke (DelegatingMethodAccessorImpl.java:43), invoke (Method.java:498), run (MainExecutor.java:37), run (Thread.java:748)
[01:31:02] [stderr] java.lang.NullPointerException
[01:31:02] [stderr]     at com.biglybt.ui.swt.systray.TrayItemDork.setMenu(TrayItemDork.java:112)
[01:31:02] [stderr]     at com.biglybt.ui.swt.systray.SystemTraySWT.<init>(SystemTraySWT.java:174)
[01:31:02] [stderr]     at com.biglybt.ui.swt.systray.SystemTraySWT.getTray(SystemTraySWT.java:82)
[01:31:02] [stderr]     at com.biglybt.ui.swt.shells.main.MainWindowImpl.showMainWindow(MainWindowImpl.java:1361)
[01:31:02] [stderr]     at com.biglybt.ui.swt.shells.main.MainWindowImpl.createWindow(MainWindowImpl.java:750)
[01:31:02] [stderr]     at com.biglybt.ui.swt.shells.main.MainWindowImpl.access$2(MainWindowImpl.java:437)
[01:31:02] [stderr]     at com.biglybt.ui.swt.shells.main.MainWindowImpl$2.runWithDisplay(MainWindowImpl.java:230)
[01:31:02] [stderr]     at com.biglybt.ui.swt.utils.SWTRunnable.run(SWTRunnable.java:37)
[01:31:02] [stderr]     at com.biglybt.ui.swt.Utils.execSWTThread(Utils.java:971)
[01:31:02] [stderr]     at com.biglybt.ui.swt.Utils.execSWTThread(Utils.java:1103)
[01:31:02] [stderr]     at com.biglybt.ui.swt.shells.main.MainWindowImpl.<init>(MainWindowImpl.java:225)
[01:31:02] [stderr]     at com.biglybt.ui.swt.shells.main.MainWindowFactory.createAsync(MainWindowFactory.java:62)
[01:31:02] [stderr]     at com.biglybt.ui.swt.Initializer.runInSWTThread(Initializer.java:277)
[01:31:02] [stderr]     at com.biglybt.ui.swt.mainwindow.SWTThread.<init>(SWTThread.java:270)
[01:31:02] [stderr]     at com.biglybt.ui.swt.mainwindow.SWTThread.createInstance(SWTThread.java:64)
[01:31:02] [stderr]     at com.biglybt.ui.swt.Initializer.<init>(Initializer.java:179)
[01:31:02] [stderr]     at com.biglybt.ui.swt.UI.takeMainThread(UI.java:110)
[01:31:02] [stderr]     at com.biglybt.ui.Main.main(Main.java:236)
[01:31:02] [stderr]     at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[01:31:02] [stderr]     at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[01:31:02] [stderr]     at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[01:31:02] [stderr]     at java.lang.reflect.Method.invoke(Method.java:498)
[01:31:02] [stderr]     at com.biglybt.launcher.MainExecutor$1.run(MainExecutor.java:37)
[01:31:02] [stderr]     at java.lang.Thread.run(Thread.java:748)
```

**Comment by ferdnyc** [comment]
I was _just_ coming here to comment, that at some point recently my tray icon came back to life! I just nonticed it for the first time, running `1.7.0.1_B44`. \o/ (It may have returned earlier than that beta, I'm slow on the uptake sometimes. I also just restarted the machine, and therefore BiglyBT, for the first time in a few days.)

But, yes, it looks like the new support is working well in LXDE (without me having to force anything). Thanks!
